core/src/main/scala/kafka/server/AutoTopicCreationManager.scala

```diff
@@ -112,6 +131,33 @@ class DefaultAutoTopicCreationManager(
     }
   }
 
+  override def getTopicCreationErrors(
+    topicNames: Set[String]
+  ): Map[String, String] = {
+    val currentTime = time.milliseconds()
+    val errors = mutable.Map.empty[String, String]
+    val expiredKeys = mutable.Set.empty[String]
+    
+    // Check requested topics and collect expired keys
+    topicNames.foreach { topicName =>
+      Option(topicCreationErrorCache.get(topicName)) match {
+        case Some(cachedError) if (currentTime - cachedError.timestamp) <= errorCacheTtlMs =>
+          errors.put(topicName, cachedError.errorMessage)
+        case Some(_) =>
+          expiredKeys += topicName
+        case None =>
+      }
+    }
+    
+    // Remove expired entries
+    expiredKeys.foreach { key =>
+      topicCreationErrorCache.remove(key)
+      debug(s"Removed expired topic creation error cache entry for $key")
+    }
+    
```

Comment:
I think we still need to make sure that eventually, errors are evicted even if we do not receive another topic creation request. I wonder if we shouldn't build an expiring cache for that implements `org.apache.kafka.common.cache.Cache`. It could use a LinkedHashMap similar LRUCache and every time we insert or get from the cache, we try to expire the last elements of the LinkedHashMap. WDYT?